test/e2e/base_user_test.go

```diff
@@ -130,7 +130,29 @@ func (s *baseUserIntegrationTest) deactivateAndCheckUser(userSignup *toolchainv1
 	require.True(s.T(), states.Deactivated(userSignup), "usersignup should be deactivated")
 }
 
-func (s *baseUserIntegrationTest) reactivateAndCheckUser(userSignup *toolchainv1alpha1.UserSignup, mur *toolchainv1alpha1.MasterUserRecord) {
+func (s *baseUserIntegrationTest) reactivateAndRequirePhoneVerification(userSignup *toolchainv1alpha1.UserSignup, mur *toolchainv1alpha1.MasterUserRecord) {
+	err := s.hostAwait.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: userSignup.Namespace,
+		Name:      userSignup.Name,
+	}, userSignup)
+	require.NoError(s.T(), err)
+
+	userSignup, err = s.hostAwait.UpdateUserSignupSpec(userSignup.Name, func(us *toolchainv1alpha1.UserSignup) {
+		states.SetDeactivating(us, false)
+		states.SetDeactivated(us, false)
+		states.SetVerificationRequired(us, true)
+	})
+	require.NoError(s.T(), err)
+	s.T().Logf("user signup '%s' reactivated, but requiring phone verification", userSignup.Name)
+
+	userSignup, err = s.hostAwait.WaitForUserSignup(userSignup.Name,
+		wait.UntilUserSignupHasConditions(ReactivatedAndVerificationRequired()...),
+		wait.UntilUserSignupHasStateLabel(toolchainv1alpha1.UserSignupStateLabelValueNotReady))
+	require.NoError(s.T(), err)
+	require.False(s.T(), states.Deactivated(userSignup), "usersignup should not be deactivated")
```

Comment:
can you check that the `userSignup` you get after calling `s.hostAwait.WaitForUserSignup()` is the same (ie, same UID) as the one passed as an arg of the reactivateAndRequirePhoneVerification?
(at this stage, we can expect that the `activation-counter` annotation was incremented, I believe)

Reply:
I'm sorry, but I don't follow what you are asking for

Reply:
> can you check that the userSignup you get after calling s.hostAwait.WaitForUserSignup() is the same (ie, same UID) as the one passed as an arg of the reactivateAndRequirePhoneVerification?

ok, now I probably know what you meant, but it doesn't make any sense to me why I should check that. I used the name of the UserSignup, so why should it be different?

> (at this stage, we can expect that the activation-counter annotation was incremented, I believe)

not yet - this user was reactivated, but it still requires phone verification. In other words, the UserSignup hasn't been approved yet, so the number cannot be incremented.